gbpservice/nfp/modules/configurator.py

```diff
@@ -0,0 +1,458 @@
+
+        :param service_type: firewall/vpn/loadbalancer/generic_config
+
+        Returns: Instance of service agent for a given service type
+
+        """
+
+        return self.cm.service_agent_instances[service_type]
+
+    def _send_request(self, operation, request_data):
+        """Maps and invokes an RPC call to a service agent method.
+
+        Takes help of de-multiplexer to get service type and corresponding
+        data and invokes the method of service agent. Service agent instance
+        is identified based on the service type passed in the request data
+
+        :param operation: Operation type - create/delete/update
+        :param request_data: RPC data
+
+        Returns: None
+
+        """
+
+        # Retrieves service type from RPC data
+        service_type = self.demuxer.get_service_type(request_data)
+        if (constants.invalid_service_type == service_type):
+            msg = ("Configurator received invalid service type %s." %
+                   service_type)
+            raise Exception(msg)
+
+        # Retrieves service agent information from RPC data
+        # Format of sa_info_list:
+        # [{'method': <m1>, 'kwargs': <rpc_data1>}, {}, ... ]
+        sa_info_list = self.demuxer.get_service_agent_info(
+                                                    operation,
+                                                    service_type,
+                                                    request_data)
+        if not sa_info_list:
+            msg = ("Configurator received invalid data format for service"
+                   " type %s. Data format: %r" % (service_type, request_data))
+            raise Exception(msg)
+
+        # Retrieves service agent instance using service type
+        sa_obj = self._get_service_agent_instance(service_type)
+        if not sa_obj:
+            msg = ("Failed to find agent with service type %s." % service_type)
+            raise Exception(msg)
+
+        # Notification data list that needs to be returned after processing
+        # RPC request. Format of notification data:
+        # notification_data[
+        #    {
+        #        'receiver': <neutron/orchestrator>,
+        #        'resource': <firewall/vpn/loadbalancer/healthmonitor/
+        #                    routes/interfaces>,
+        #        'method': <network_function_device_notification/
+        #                  *aaS response RPC method name>,
+        #        'kwargs': [{<data1>}, {data2}]
+        #    },
+        #    {
+        #    }, ...
+        # ]
+        #
+        # Initially, notification data will be empty and is populated
+        # after processing each request data in the request data list
+        notification_data = {}
+
+        # Handover the request data list and notification data to the
+        # identified service agent
+        sa_obj.forward_request(sa_info_list, notification_data)
+
+    def create_network_device_config(self, context, request_data):
```

Comment:
Sync up with Magesh, be consistent with the names in orchestrator. network_device is network_function_device. If this is too long you can use network_device.